lib/runner/variant-reassignment.js

```diff
@@ -332,10 +333,44 @@ export default class VariantReassignment {
 
   /**
    * Verify that there are no other products in the platform that has slugs of productDraft
-   * except ctpProductToUpdate.
-   * This method should cover the test variant-reassignment-example-7.spec.js
+   * except ctpProductToUpdate. It's enough to check matchingProducts because it's not probable
+   * that there will be a product which has no matching variant, but a conflicting slug.
+   *
+   * @see variant-reassignment-example-7.spec.js
+   */
+  async _ensureSlugUniqueness (productDraft, matchingProducts) {
+    const productDraftSlug = productDraft.slug
+    const productsToAnonymize = []
+    matchingProducts.forEach((product) => {
+      const isSlugConflicting = this._isSlugConflicting(product, productDraftSlug)
+      if (isSlugConflicting)
+        productsToAnonymize.push(product)
+    })
+
+    await Promise.map(productsToAnonymize, product =>
+        this.productService.anonymizeCtpProduct(product)
+      , { concurrency: 3 })
+  }
+
+  /**
+   * The slugs from product and product draft are conflicting
+   * when at least one language from product's slug is the same as in product draft slug
+   * @param product
+   * @param productDraftSlug
+   * @returns {boolean}
+   * @private
    */
-  _ensureSlugUniqueness () {
+  _isSlugConflicting (product, productDraftSlug) {
+    const stagedSlug = product.masterData.staged.slug
+    const currentSlug = product.masterData.current.slug
+    const stagedDraftSlugs = _.merge({}, productDraftSlug, stagedSlug)
+    const currentDraftSlugs = _.merge({}, productDraftSlug, currentSlug)
+    const productDraftSlugLength = Object.keys(productDraftSlug).length
+    const stagedSlugLength = Object.keys(stagedSlug).length
+    const currentSlugLength = Object.keys(currentSlug).length
+
+    return stagedSlugLength + productDraftSlugLength > Object.keys(stagedDraftSlugs).length
+      || currentSlugLength + productDraftSlugLength > Object.keys(currentDraftSlugs).length
```

Comment:
I tried to clear this method using loop:
```js
_isSlugConflicting (product, productDraftSlug) {
  const productDraftSlugLength = Object.keys(productDraftSlug).length
  
  // if at least one version has conflict in slugs, return true
  for(let version of ['staged', 'current']) {
    const slug = product.masterData[version].slug
    const slugLength = Object.keys(stagedSlug).length
    const stagedDraftSlugs = _.merge({}, productDraftSlug, slug)

    if(slugLength + productDraftSlugLength !== Object.keys(stagedDraftSlugs).length)
      return true
  }
  return false
}
```

Reply:
https://github.com/commercetools/commercetools-node-variant-reassignment/pull/37/commits/626d763a9fae27de8b2035fd1857ca6a8346856b